custom_components/amplipi/media_player.py

```diff
@@ -475,7 +482,8 @@ def source(self):
     def source_list(self):
         """List of available input sources."""
         streams = ['None']
-        streams += [stream.name for stream in self._streams if stream.id >= 1000 or stream.id - 996 == self._id]
+        if self._streams is not None:
+            streams += [stream.name for stream in self._streams if stream.id >= 1000 or stream.id - 996 == self._id]
```

Comment:
You may want to explain why this is hiding RCAs. Is something like this easier to read?
```suggestion
            not_selectable = [996, 997, 998, 999]
            rca_selectable = RCAs.get(self.id)
            streams += [stream.name for stream in self._streams if stream.id not in RCAs or stream.id == rca_selectable]
```

Reply:
Here's what I came up with after editing your code somewhat:
```
    @property
    def source_list(self):
        """List of available input sources."""
        streams = ['None']
        if self._streams is not None:
            # Excludes every RCA except for the one related to the given source
            RCAs = [996, 997, 998, 999]
            rca_selectable = RCAs[self._id]
            streams += [stream.name for stream in self._streams if stream.id not in RCAs or stream.id == rca_selectable]
        return streams
```